File: ios/SampleUI/Sources/SampleUI/Model/UseSmileIDSampleScenario.swift
/// The flow scenarios the drawer offers, asserted against `spec/scenarios.json` by a unit test.
public enum UseSmileIDSampleScenario: String, CaseIterable, Sendable {
  case normal
  case expiredToken
  case badRefresh
  case noCallback
  case throwingCallback
  case offlineRetry

  public var id: String {
    rawValue
  }

  public var label: String {
    switch self {
    case .normal: "Normal"
    case .expiredToken: "Expired token"
    case .badRefresh: "Refresh fails"
    case .noCallback: "No result callback"
    case .throwingCallback: "Throwing callback"
    case .offlineRetry: "Offline then retry"
    }
  }

  public var description: String {
    switch self {
    case .normal: "Happy path with valid sandbox credentials."
    case .expiredToken: "Token is valid but expired, so the SDK must refresh before it can submit."
    case .badRefresh: "Refresh returns an unusable token, so the failure path must surface."
    case .noCallback: "Host provides no result callback; the SDK must not crash or hang."
    case .throwingCallback: "Host callback throws; it must not corrupt SDK state."
    case .offlineRetry: "Submission starts with no connectivity, then it returns."
    }
  }

  /// Restored identifiers fall back rather than throw, so a retired scenario cannot crash a restore.
  public init(id: String?) {
    self = UseSmileIDSampleScenario(rawValue: id ?? "") ?? .normal
  }
}

/// Theme scenarios apply on top of any flow scenario, through the SDK's public theme override.
public enum UseSmileIDSampleThemeScenario: String, CaseIterable, Sendable {
  case brandDefault
  case clashingHost
  case partnerOverride

  public var id: String {
    rawValue
  }

  public var label: String {
    switch self {
    case .brandDefault: "Brand default"
    case .clashingHost: "Clashing host"
    case .partnerOverride: "Partner override"
    }
  }

  public var description: String {
    switch self {
    case .brandDefault: "Ship state: Smile ID branding, light or dark per the Settings switch."
    case .clashingHost: "A deliberately distant host theme, to expose host-versus-SDK collisions."
    case .partnerOverride: "A plausible partner palette through the same public override."
    }
  }

  public init(id: String?) {
    self = UseSmileIDSampleThemeScenario(rawValue: id ?? "") ?? .brandDefault
  }
}

File: expo/sample-ui/src/model/use-smile-id-sample-scenario.ts
/// One scenario the drawer offers, with the copy the drawer and the result card both read.
export type UseSmileIDSampleScenarioEntry = {
  readonly id: string;
  readonly label: string;
  readonly description: string;
};

/// The flow scenarios the drawer offers, asserted against `spec/scenarios.json` by a unit test.
export const smileIDSampleScenarios: readonly UseSmileIDSampleScenarioEntry[] = [
  { id: 'normal', label: 'Normal', description: 'Happy path with valid sandbox credentials.' },
  {
    id: 'expiredToken',
    label: 'Expired token',
    description: 'Token is valid but expired, so the SDK must refresh before it can submit.',
  },
  {
    id: 'badRefresh',
    label: 'Refresh fails',
    description: 'Refresh returns an unusable token, so the failure path must surface.',
  },
  {
    id: 'noCallback',
    label: 'No result callback',
    description: 'Host provides no result callback; the SDK must not crash or hang.',
  },
  {
    id: 'throwingCallback',
    label: 'Throwing callback',
    description: 'Host callback throws; it must not corrupt SDK state.',
  },
  {
    id: 'offlineRetry',
    label: 'Offline then retry',
    description: 'Submission starts with no connectivity, then it returns.',
  },
] as const;

/// Theme scenarios apply on top of any flow scenario, through the SDK's public theme override.
export const smileIDSampleThemeScenarios: readonly UseSmileIDSampleScenarioEntry[] = [
  {
    id: 'brandDefault',
    label: 'Brand default',
    description: 'Ship state: Smile ID branding, light or dark per the Settings switch.',
  },
  {
    id: 'clashingHost',
    label: 'Clashing host',
    description: 'A deliberately distant host theme, to expose host-versus-SDK collisions.',
  },
  {
    id: 'partnerOverride',
    label: 'Partner override',
    description: 'A plausible partner palette through the same public override.',
  },
] as const;

/// The SDK flow is one route with two presentations; same destination, different container.
export const smileIDSampleFlowRoutes = ['fullscreen', 'shell'] as const;

export type UseSmileIDSampleFlowRoute = (typeof smileIDSampleFlowRoutes)[number];
